### PR TITLE
Will-Ratterman-JS-Basics

### PR DESCRIPTION
**Pull request by wratterman**

### CODE REVIEW

File: lib/triangles/is-triangle.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+function isRightTriangle (side1, side2, side3) {
+  if (side1 <= 0 || side2 <= 0 || side3 <= 0){
+        return false;
+    } else if ((Math.pow(side1, 2) + Math.pow(side2, 2)) === Math.pow(side3, 2)) {
```
Comment (case-eee) [inline lib/triangles/is-triangle.js]: you've got some repeated logic here. it doesn't matter what order you add the first two numbers together, you don't need to do it twice 👍

File: test/averages/mean-test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const assert = require('chai').assert
+const mean = require('../../lib/averages/mean.js')
+
+describe('mean functionality', function() {
+  context('mean function', function(){
+    it('it can determine the average of an array of numbers', function(){
```
Comment (case-eee) [inline test/averages/mean-test.js]: you have a lot of use cases all within this it block. i suggest creating a different it block for each use case. for example...

```js
it('returns the average of an array of integers', function(){
...
}
it('returns null if the array is empty', function(){
...
}

```

File: test/factorials/factorial-test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const assert = require('chai').assert
+const factorialCalculator = require('../../lib/factorials/factorial.js')
+
+describe('factorialCalculator functionality', function() {
+  context('factorialCalculator function', function(){
+    it('it can calculate the factorial of a number', function(){
+      let number = 0;
```
Comment (case-eee) [inline test/factorials/factorial-test.js]: same here. you are trying quite a few inputs which is a good in theory, but not necessary. i suggest only testing the edge cases along with one "normal" case - probably test 0 and also any integer.
